File: Module 03 - Back-end Development and Git Github introduction/challenge-02/src/routes/place_routes.ts
import { Router } from "express";
import { createPlace, deletePlace, getPlaceById, getPlaces, updatePlace } from "../controllers/places_controller"
const placeRouter = Router();


placeRouter.post("/", createPlace);
placeRouter.get("/all", getPlaces);
placeRouter.get("/unique/:place_id", getPlaceById);
placeRouter.delete("/:place_id", deletePlace);
placeRouter.patch("/:place_id", updatePlace);

export { placeRouter };

File: Module 03 - Back-end Development and Git Github introduction/challenge-02-flexing/src/routes/place_routes.rs
use axum::{Router, routing::{post, get, patch, delete}, body::Body};

use crate::{controllers::place_controller::PlaceController,  models::app_data::AppDataArc};


pub struct PlaceRouter;

impl PlaceRouter {
  pub fn new() -> Router<AppDataArc, Body>{
    return Router::new()
      .route("/", post(PlaceController::create_place))
      .route("/all", get(PlaceController::get_all_places))
      .route("/unique/:place_id", get(PlaceController::get_place))
      .route("/:place_id", patch(PlaceController::update_place))
      .route("/:place_id", delete(PlaceController::delete_place))

    
  }
}